backend/coins/btc/handlers/handlers.go

```diff
@@ -189,14 +190,18 @@ func (handlers *Handlers) getTxInfoJSON(txInfo *accounts.TransactionData, detail
 		Unit:   amount.Unit,
 	}
 	var formattedTime *string
-	if txInfo.Timestamp != nil {
-		t := txInfo.Timestamp.Format(time.RFC3339)
-		formattedTime = &t
-		amountAtTime = handlers.formatAmountAtTimeAsJSON(txInfo.Amount, txInfo.Timestamp)
-	} else if txInfo.CreatedTimestamp != nil {
-		t := txInfo.CreatedTimestamp.Format(time.RFC3339)
+	var deductedAmount FormattedAmount
+	timestamp := txInfo.Timestamp
+	if timestamp == nil {
+		timestamp = txInfo.CreatedTimestamp
+	}
+	if timestamp != nil {
+		t := timestamp.Format(time.RFC3339)
 		formattedTime = &t
-		amountAtTime = handlers.formatAmountAtTimeAsJSON(txInfo.Amount, txInfo.CreatedTimestamp)
+		amountAtTime = handlers.formatAmountAtTimeAsJSON(txInfo.Amount, timestamp)
+		if txInfo.Fee != nil && txInfo.Type == accounts.TxTypeSend {
+			deductedAmount = handlers.formatAmountAtTimeAsJSON(coin.SumAmounts(txInfo.Amount, *txInfo.Fee), timestamp)
```

Comment:
@bznein @Beerosagos I don't think this works with ERC-20 tokens, where the fee has a different unit (ETH) than the ERC-20 token. See `txInfo.FeeIsDifferentUnit`.

Furthermore, the handler should ideally not contain logic, but only convert http inputs and outputs. Shouldn't this stuff be wherever TransactionData is being prepared?

Reply:
Shoot, you're right about the ERC20 think!

About the place in the code for this change, I guess you're right too :innocent: I didn't consider it a problem as I thought the deducted amount as a mere data representation used in the frontend and not useful in other parts of the backend, but I can see how it would be better to have it in TransactionData with some unit testing.

Reply:
Btw, while changing this, I think that it would be useful to add the coin type to the Tx, to make conversion estimation checks easier in the frontend cc @thisconnect